### DISCUSSION TITLE
Supporting k8s versions

### DISCUSSION THREAD

**Discussion by icarmon** [fabric8io/kubernetes-client#4169] [Q&A] [answered]
Hi all
Can anyone give me an estimate on how long it takes to support newer versions of K8s?

How dangerous is it to use the latest client on a version that's not yet officially supported?

Thanks!

**Comment by manusa** [comment, marked as answer]
Hi @icarmon 

With the newer versions of the Client (>=5.x) it shouldn't be dangerous at all.
Nowadays, the version compatibility matrix only reflects the generated Java types matching Kubernetes core resources. So when we say that K8s 1.23 is fully supported, it really means that we (hopefully) generated all of the Java types that represent the vanilla Kubernetes resources available in that version. We should have also provided DSL entry points to be able to query them.

Anyway, and regarding safety, in case we didn't provide specific support for a more modern (or outdated) K8s resource, in the newer versions any unknown resource will be _deserialized_ into a GenericKubernetesResource.

**Comment by icarmon** [comment]
Thanks @manusa 
I really appreciate the detailed response

**Comment by yelnyleenew** [comment]
I think may can give a detail mapping for us, it's too troubled ，when i need support  k8s version(1.21/1.22/1.23/1.24), I dont know which version I can selected

**Reply by manusa** [reply]
You should be fine using any of the latest Kubernetes Client versions regardless of the Kubernetes Cluster version.
